src/hivpy/sexual_behaviour.py

```diff
@@ -47,6 +48,45 @@ def __init__(self, **kwargs):
         }
         self.short_term_partners = {SexType.Male: self.sb_data.male_stp_dists,
                                     SexType.Female: self.sb_data.female_stp_u25_dists}
+        self.ltp_risk_factor = self.sb_data.rred_long_term_partnered.sample()
+        self.rred_diagnosis = self.sb_data.rred_diagnosis.sample()
+        self.rred_diagnosis_period = datetime.timedelta(days=365)*self.sb_data.rred_diagnosis_period
+        self.yearly_risk_change = {"1990s": self.sb_data.yearly_risk_change["1990s"].sample(),
+                                   "2010s": self.sb_data.yearly_risk_change["2010s"].sample()}
+        # only need to calculate rred_art_adherence in fraction of runs
+        self.rred_art_adherence_flag = (
+            rng.random() < self.sb_data.rred_art_adherence_probability)
+        self.rred_art_adherence = self.sb_data.rred_art_adherence
+        self.adherence_threshold = self.sb_data.adherence_threshold
+        self.new_partner_factor = self.sb_data.new_partner_dist.sample()
+
+    def age_index(self, age):
+        return np.minimum((age.astype(int)-self.risk_min_age) //
+                          self.risk_age_grouping, self.risk_categories)
+
+    def update_sex_behaviour(self, population):
+        self.num_short_term_partners(population.data)
+        self.update_sex_groups(population.data)
+        self.update_rred(population)
+
+    def update_rred(self, population):
```

Comment:
update_rred uses population data and date, so it seemed simpler to me to pass in the whole population object so that there wouldn't be a chance of them being out of sync (e.g. using a date that wasn't associated with that population). 

I'm not sure if passing around population / population data like this is the best approach anyway though, since I think a sexual behaviour module is only going to be associated with one population, we simplify this by just having a member variable point to that population. But then I think the population object has a sexual behaviour module member variable so it starts feeling a bit circular.